metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/bolt/ParserBolt.java

```diff
@@ -182,40 +185,61 @@ public void prepare(Map stormConf, TopologyContext context, OutputCollector coll
     super.prepare(stormConf, context, collector);
     messageGetStrategy = MessageGetters.DEFAULT_BYTES_FROM_POSITION.get();
     this.collector = collector;
-    if(getSensorParserConfig() != null) {
-      cache = CachingStellarProcessor.createCache(getSensorParserConfig().getCacheConfig());
-    }
-    initializeStellar();
-    if(getSensorParserConfig() != null && filter == null) {
-      getSensorParserConfig().getParserConfig().putIfAbsent("stellarContext", stellarContext);
-      if (!StringUtils.isEmpty(getSensorParserConfig().getFilterClassName())) {
-        filter = Filters.get(getSensorParserConfig().getFilterClassName()
-                , getSensorParserConfig().getParserConfig()
-        );
+
+    // Build the Stellar cache
+    Map<String, Object> cacheConfig = new HashMap<>();
+    for (Map.Entry<String, ParserComponents> entry: sensorToComponentMap.entrySet()) {
+      String sensor = entry.getKey();
+      SensorParserConfig config = getSensorParserConfig(sensor);
+
+      if (config != null) {
+        cacheConfig.putAll(config.getCacheConfig());
       }
     }
+    cache = CachingStellarProcessor.createCache(cacheConfig);
 
-    parser.init();
+    // Need to prep all sensors
+    for (Map.Entry<String, ParserComponents> entry: sensorToComponentMap.entrySet()) {
+      String sensor = entry.getKey();
+      MessageParser<JSONObject> parser = entry.getValue().getMessageParser();
 
```

Comment:
I don't believe this is correct.  We want to initialize stellar PER parser.  Each should have it's own stellar instance and cache.

Reply:
I left it as a single shared cache on purpose. 

I don't believe that there'd be any incorrect evictions by sharing the cache, and I think evicting based on the overall usage in the aggregated parser is the appropriate place to handle it. Since the cache is (mostly) LRU, I'd prefer to drop the least recently used entry of all parsers rather than dropping for each parser.  LRU of the overall flow seems better than LRU of each of the sensors. Assuming a single cache, you'd bump up the cache configs to account for this, rather than having to optimize each config individually (and potentially as a group afterwards).

Caching is also off by default for the parsers, so this is a case that's only hit if the user explicitly chooses to do so.

Having said that, I do think I need to shore up the documentation around that logic, assuming we choose to go forward with it.  Let me know what you think, and I can adjust appropriately.

Reply:
We may need a test then.

Reply:
I'm still not sure sharing the stellar Context is a good thing. @cestella

Reply:
So, the consequences of this decision are as follows:
* You share an expression cache (i.e. the statement -> abstract syntax tree cache; distinct from the expression -> evaluated return cache)
* You share an stellar value cache (expression -> evaluated return)
* You share the state in the Context (e.g. hbase connections, zookeeper connections).

On the whole, anything shared in the context is intended to be shared across users and sensors by virtue of Stellar being used in the enrichment topology (where it's not sensor-by-sensor), so we shoudl be ok there.  The real question is whether users would prefer to have one knob per topology for stellar cache sizing or whether they would prefer to have one knob per sensor.  I'd say that I'm ok with how this PR is doing it, because it's easier to reason about resources, IMO, on a per-topology perspective.